src/Symfony/Component/Serializer/DependencyInjection/SerializerPass.php

```diff
@@ -58,6 +58,7 @@ public function process(ContainerBuilder $container): void
             $defaultContext = $container->getParameter('serializer.default_context');
             $this->bindDefaultContext($container, array_merge($normalizers, $encoders), $defaultContext);
             $container->getParameterBag()->remove('serializer.default_context');
+            $container->getDefinition('serializer')->setArgument(2, $defaultContext);
```

Comment:
I would have set
```suggestion
            $container->getDefinition('serializer')->replaceArgument(2, $defaultContext);
```
 And updated the definition in `serializer.php`